app/src/main/java/de/westnordost/streetcomplete/quests/SharedFilters.kt

```diff
@@ -0,0 +1,12 @@
+package de.westnordost.streetcomplete.quests
+
+import de.westnordost.streetcomplete.osm.MAXSPEED_TYPE_KEYS
+
+val anyMaxSpeedTagKey = "~${(MAXSPEED_TYPE_KEYS + "maxspeed").joinToString("|")}"
+
+val onlyInANoneMaxSpeed30OrLessZone = """
+  (
+    $anyMaxSpeedTagKey !~ ".*:(zone)?:?([1-9]|[1-2][0-9]|30)"
+    and $anyMaxSpeedTagKey ~ ".*:(urban|rural|trunk|motorway|nsl_single|nsl_dual)"
```

Comment:
I thought a reason to do this at all was to get rid of the hardcoded list of maxspeed type values (since it can be somewhat of a free-form value depending on what the local community decides what values there should be, see GB community with their nsl stuff.

Reply:
The only other thing I can think of right now that works with all unit tests is the following:

```
val onlyInANoneMaxSpeed30OrLessZone = """
  (
    $anyMaxSpeedTagKey
    and !(maxspeed ~ "walk|(\d+( ?mph)?)" and ${(MAXSPEED_TYPE_KEYS).joinToString(" and ") { "!$it" }})
    and $anyMaxSpeedTagKey !~ ".*:(zone)?:?([1-9]|[1-2][0-9]|30)"
  )
"""
```

Reply:
Why wouldn't `$anyMaxSpeedTagKey ~ ".*:(zone)?:?([1-9]|[1-2][0-9]|30)"` suffice?

Reply:
Because it doesn't capture the assumption that something that is only tagged with the `maxspeed` tag and no other maxspeed tags is assumed to be in a max-speed zone if the speed is low enough.

With following it works and is more explicit:
```
  !(
    maxspeed ~ "walk|(\d+( ?mph)?)" and !(~${(MAXSPEED_TYPE_KEYS).joinToString("|")})
    or $explicitlyInSlowZone
  )
```

Reply:
If I leave out the part about having only the `maxspeed` tag and no other maxspeed tags, the following test fails. That means the quest would be shown for more elements than desired:

https://github.com/streetcomplete/StreetComplete/blob/e300eaab151c413c387e9a9e936e2b9013ad5dab/app/src/test/java/de/westnordost/streetcomplete/quests/lanes/AddLanesTest.kt#L55-L59